internal/tjson/schema.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tjson
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+// SchemaType represents JSON value type in JSON Schema.
+type SchemaType string
+
+// JSON value types defined by the JSON Schema.
+const (
+	Integer SchemaType = "integer"
+	Number  SchemaType = "number"
+	String  SchemaType = "string"
+	Boolean SchemaType = "boolean"
+	Array   SchemaType = "array"
+	Object  SchemaType = "object"
+)
+
+// SchemaFormat represents additional information about JSON value type in JSON Schema.
+type SchemaFormat string
+
+// JSON value formats.
+const (
+	EmptyFormat SchemaFormat = ""
+
+	// For Number.
+	Double SchemaFormat = "double"
```

Comment:
Is it connected somehow to the Tigris internal representation? 
If not then would you please add a comment what is the difference between float and double values? 
What for the `FloatSchema` used?